### ISSUE TITLE
Add OpenTelemetry logging support

### ISSUE THREAD

**Issue by BryanHuntNV** [loglayer/loglayer#45]
Could we get OpenTelemetry supported as a logger?

**Comment by theogravity** [comment]
Can you write example code that uses open telemetry showing how I would do logging? Docs aren't clear on how to do logging with it.

If you can provide an example I can spend time today looking into it since I have the day off.

**Comment by BryanHuntNV** [comment]
This is the best starting point I can provide right now: https://github.com/open-telemetry/opentelemetry-js-contrib/blob/main/plugins/node/opentelemetry-instrumentation-winston/src/instrumentation.ts

It is rather convoluted to read, but hopefully it would take you down the right path.  If you don't get anywhere with that code, I can try to dig deeper next week.

**Comment by theogravity** [comment]
Are you using winston? If so, why not just attach it to winston and use winston with loglayer?

My familiarity with these kinds of observational libs is with DataDog and we use pino with loglayer and i use a plugin callback on loglayer to inject the trace id before the log ships to pino.

**Comment by BryanHuntNV** [comment]
I'd be ok closing the request.

**Comment by theogravity** [comment]
To me, they're two separate domains which is why I needed an example on how one would do logging with the lib itself.

**Comment by theogravity** [comment]
> I'd be ok closing the request.

Yeah unfortunately I don't think opentelemetry falls under something that loglayer would support since it's not really a logging lib.

**Comment by BryanHuntNV** [comment]
FWIW: https://www.perplexity.ai/search/can-i-do-direct-logging-in-ope-HFxWqg5jQgKnaYXXwI3edA

**Comment by BryanHuntNV** [comment]
I won't have a problem if you close this.  Sometimes my ideas are bad.

**Comment by theogravity** [comment]
That is the most messiest API for logging I've seen and it's clear that the lib wasn't meant to do that. 

If your goal is to reduce dependencies, why install a whole slew of them to just enable logging for it vs using a dedicated logging library?

Is it possible now that you've provided the API? Yes, but it makes me really hesitant to want to do it.

But also gives me ideas on for a next major version of loglayer where i'd expose a plugin system for shipping logs.

**Comment by theogravity** [comment]
Let's keep this open for now. I do want to work on a new major version that allows for multiple loggers and the ability to write your own shippers.

**Comment by theogravity** [comment]
> I won't have a problem if you close this.  Sometimes my ideas are bad.

Yeah, it's a bad idea, but you gave some inspiration!

**Comment by theogravity** [comment]
LogLayer 5.x is now out which should now support adding something like this. You can do it by creating your own [Transport](https://loglayer.dev/transports/creating-transports.html) where you'd assign the incoming message parameters to the log entry on the OpenTelemetry side.

It's not something I think I'd want to make into an official plugin because it still feels strange as a concept to me, but if you have a working sample, I can def consider adding it for others to use!

**Comment by theogravity** [comment]
I've added support for OpenTelemetry. See docs for more info.
